cloudferrylib/scheduler/scenario.py

```diff
@@ -117,3 +123,84 @@ def init_process_migrate(self, path):
         process = migrate['process']
         namespace = migrate['namespace']
         return process, namespace
+
+
+class ScenarioChecker():
+    def __init__(self, tasks, key, scenario):
+        self.tasks = tasks
+        self.key = key
+        self.scenario = scenario
+        self.missed_tasks = set()
+        self.duplicated_tasks = set()
+        self.missed_links = set()
+        self.links = set()
+        self.active_tasks = set()
+
+    def check(self):
+        self.check_dict_list(self.scenario)
+
+        # check missed links
+        for link in self.links:
+            if link not in self.active_tasks:
+                self.missed_links.add(link)
+
+        throw_exception = False
+        # log all errors if needed
+        if len(self.missed_links):
+            LOG.error("Scenario has missed links: %s",
+                      ", ".join(self.missed_links))
+            throw_exception = True
+
+        if len(self.duplicated_tasks):
+            LOG.error("Scenario has duplicated tasks: %s",
+                      ", ".join(self.duplicated_tasks))
+            throw_exception = True
+
+        if len(self.missed_tasks):
+            LOG.error("Following tasks not described in tasks.yaml: %s",
+                      ", ".join(self.missed_tasks))
+            throw_exception = True
+
+        if throw_exception:
+            raise ScenarioError(self)
+
+    def check_string_list(self, node_list):
+        for node in node_list:
+            self.links.add(node)
+
+    def check_node(self, node):
+        key, value = node
+        if type(value) is type(list()):
```

Comment:
Maybe it is better to have `if isinstance(value, list):` instead of types compare?